src/System.Text.Primitives/System/Text/Encoding/Utf8/Utf8Encoder.cs

```diff
@@ -56,6 +56,25 @@ public static bool TryDecode(ReadOnlySpan<byte> utf8, Span<char> utf16, out int
             bytesConsumed = 0;
             charactersWritten = 0;
 
+            if (Vector.IsHardwareAccelerated)
+            {
+                var utf8Vectors = utf8.NonPortableCast<byte, Vector<byte>>();
+                
+                int asciiLoops = Math.Min(utf8.Length, utf16.Length) / Vector<byte>.Count, asciiLoop = 0, charOffset = 0;
+                while (asciiLoops-- != 0 && (utf8Vectors[asciiLoop] & _highBit) == Vector<byte>.Zero)
```

Comment:
Using the span indexer is slow. After the NonPortableCast call:
ref Vector<byte> vectors = utf8Vectors.DangerousGetPinnableReference();

then anywhere you need a value:
Unsafe.Add(ref vectors, idx) = <blah>;   // <= This works for get or set.

This will be much faster and compiles down to almost identical code that would have been if you had used fixed block and pointer.